### PR TITLE
refactor(on-ramp): refactor quotes screen

### PR DESCRIPTION
**Pull request by wachunei**
**Development & PR Process**
1. Follow MetaMask [Mobile Coding Standards](https://docs.google.com/document/d/1VJLwTRsUw_5EDq_o8d6sSbXUAYENLurkRitYO45eY5o/edit?usp=sharing)
2. Add `release-xx` label to identify the PR slated for a upcoming release (will be used in release discussion)
3. Add `needs-dev-review` label when work is completed
4. Add `needs-qa` label when dev review is completed
5. Add `QA Passed` label when QA has signed off

**Description**

This PR refactors the Quotes screen of the on-ramp with a new design and file structure.

**Screenshots/Recordings**

Before:

https://github.com/MetaMask/metamask-mobile/assets/1024246/47b8bc73-9c4d-4ea5-aff6-b3c2fab7c7da

After:

https://github.com/MetaMask/metamask-mobile/assets/1024246/b30af963-4e9f-4d96-84c3-0c425045f080

**Issue**

Progresses #???

**Checklist**

* [ ] There is a related GitHub issue
* [ ] Tests are included if applicable
* [ ] Any added code is fully documented

### CODE REVIEW

**Comment by wachunei** [comment]
@SocketSecurity ignore-all

**Review by alokeutpal — APPROVED** [review]

File: app/components/UI/FiatOnRampAggregator/Views/Quotes/Quotes.constants.ts [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+
+import { QuoteError, QuoteResponse } from '@consensys/on-ramp-sdk';
+import { DeepPartial } from './Quotes.types';
+
+export const mockQuotesData = [
+  {
+    provider: {
+      id: '/providers/banxa-staging',
+      name: 'Banxa (Staging)',
+      description:
```
Comment (sethkfman) [inline app/components/UI/FiatOnRampAggregator/Views/Quotes/Quotes.constants.ts:11]: Is there a reason why none of this is using `strings()` to enable translations?

**Review by sethkfman — APPROVED** [review]
LGTM

**Comment by bkirb** [comment]
@wachunei LGTM I finished verifying this change

**Review by sethkfman — DISMISSED** [review]
LGTM

**Review by sethkfman — DISMISSED** [review]

**Review by sethkfman — DISMISSED** [review]

**Comment by wachunei** [comment]
Closed in favor of https://github.com/MetaMask/metamask-mobile/pull/6804

**Comment by sethkfman** [comment]
Closed due to CI not updating. Another ticket was opened but this provides the review history.

**Pull request closed without merging**